### PR TITLE
Rabbitmq

### PR DESCRIPTION
**Pull request by ghost**
Includes SSE

### CODE REVIEW

**Review by jusoto — APPROVED** [review]

**Review by dmssargent — APPROVED** [review]

**Comment by dmssargent** [comment]
The streams in the current PR are made on demand per routing key, not per user id. That makes nearly impossible for the frontend to bind to the notification service.

File: internal/rabbitMQ/queuedeclaration.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package rabbitMQ
+
+import (
+	"log"
+
+	"github.com/streadway/amqp"
+	"github.com/pkg/errors"
+)
+
+func QueueDeclarations (ch *amqp.Channel, err error) (amqp.Queue, error) {
```
Comment (dmssargent) [inline internal/rabbitMQ/queuedeclaration.go:10]: nit picky: space after function name

File: cmd/notification-service/main.go [hunk 2]
```diff
@@ -2,66 +2,55 @@ package main
...
 	go func() {
-		server.Publish("messages", &sse.Event{
-			Data:  []byte("beat"),
-			Event: []byte("1"),
-		})
-	}()
+		rabbitMQ.InitializeConsumer(func(m amqp.Delivery){
+			log.Printf("Received message")
 
+			data, err := json.Marshal(model.Message{RoutingKey: m.RoutingKey, Body: m.Body})
```
Comment (dmssargent) [inline cmd/notification-service/main.go]: m.Body has type `[]byte` which per https://golang.org/pkg/encoding/json/#Marshal gets encoded by the rule "...[]byte encodes as a base64-encoded string" which is annoying. Can we normalize the []byte to a string?

File: cmd/notification-service/main.go [hunk 3]
```diff
@@ -2,66 +2,55 @@ package main
...
 
+			if err != nil {
+				log.Printf("Parsing json failed")
+				return
+			}
+
+			server.CreateStream(m.UserId)
```
Comment (dmssargent) [inline cmd/notification-service/main.go]: The userId of the message is per Rabbit not per the system we are building. The correct user id should be in a field of the JSON message we should be receiving from RabbitMQ.

**Review by dmssargent — CHANGES_REQUESTED** [review]

**Pull request merged**